servers/src/main/java/tachyon/worker/file/FileWorkerMasterSyncExecutor.java

```diff
@@ -97,8 +96,23 @@ public void heartbeat() {
     }
 
     for (PersistFile persistFile : command.getCommandOptions().getPersistOptions().persistFiles) {
+      long fileId = persistFile.fileId;
+      if (mFileDataManager.isFilePersisting(fileId) || mFileDataManager.isFilePersisted(fileId)) {
```

Comment:
Should there just be a single public method in FileDataManager that takes care of all of this logic [lines 100 - 109]?